# SC1015 : Introduction to Data Science and Artificial Intelligence
# Mini-Project : Spread of Covid in US Counties

In [166]:
# Basic Libraries
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt
sb.set()

In [167]:
usa_county = pd.read_csv('usa_county_wise.csv')
usa_county.head()

UID iso2 iso3  code3     FIPS    Admin2            Province_State  \
0        16   AS  ASM     16     60.0       NaN            American Samoa   
1       316   GU  GUM    316     66.0       NaN                      Guam   
2       580   MP  MNP    580     69.0       NaN  Northern Mariana Islands   
3  63072001   PR  PRI    630  72001.0  Adjuntas               Puerto Rico   
4  63072003   PR  PRI    630  72003.0    Aguada               Puerto Rico   

  Country_Region        Lat       Long_                  Combined_Key  \
0             US -14.271000 -170.132000            American Samoa, US   
1             US  13.444300  144.793700                      Guam, US   
2             US  15.097900  145.673900  Northern Mariana Islands, US   
3             US  18.180117  -66.754367     Adjuntas, Puerto Rico, US   
4             US  18.360255  -67.175131       Aguada, Puerto Rico, US   

      Date  Confirmed  Deaths  
0  1/22/20          0       0  
1  1/22/20          0       0  
2  1/22/20          0       0  
3  1/22/20          0       0  
4  1/22/20          0       0

Here, we have a dataset with over 600,000 data points. This covers all 3000+ counties over the span of over 6 months, covering number of infected and deaths by Covid per day. While very extensive, we felt that not all the data was relevant, and instead wanted to show a general trend, so we felt working with all 600,000 data points was not necessary. The code below was used to truncate the data set, filtering by a specific date, in this case 27 July 2020.

In [168]:
#usa_county_27july = usa_county[usa_county['Date']=='7/27/20']
#usa_county_27july.to_csv('usa_county_27july.csv', index=False)

Having done that, we will start importing all the relevant data sets, namely:
> Average Household Size and Population Density  
> Health Insurance Coverage  
> Income and Benefits  
> Occupation by Sex  
> Population and Poverty Status  
> USA County Covid Tracking

We felt these demographic factors might give insight to how and why Covid may have spread differently based on each county. We hope to be able to merge these data sets, as a part of our data cleaning, so that we may view the data points together.

In [169]:
housePop = pd.read_csv('Average_Household_Size_and_Population_Density_-_County.csv')
housePop.head()

COUNTYNS  Geographic Identifier - FIPS Code  Area of Land (square meters)  \
0    161526                               1001                  1.539602e+09   
1    161527                               1003                  4.117547e+09   
2    161528                               1005                  2.292145e+09   
3    161529                               1007                  1.612167e+09   
4    161530                               1009                  1.670104e+09   

   Area of Water (square meters)            Name    State  \
0                       25706961  Autauga County  Alabama   
1                     1133055836  Baldwin County  Alabama   
2                       50538698  Barbour County  Alabama   
3                        9602089     Bibb County  Alabama   
4                       15015423   Blount County  Alabama   

   Average Household Size  Average Household Size - Margin of Error  \
0                    2.59                                      0.05   
1                    2.61                                      0.04   
2                    2.49                                      0.07   
3                    2.99                                      0.14   
4                    2.77                                      0.05   

   Average Household Size of Owner-Occupied Unit  \
0                                           2.59   
1                                           2.66   
2                                           2.44   
3                                           3.05   
4                                           2.85   

   Average Household Size of Owner-Occupied Unit - Margin of Error  ...  \
0                                               0.07                ...   
1                                               0.06                ...   
2                                               0.11                ...   
3                                               0.18                ...   
4                                               0.07                ...   

   Total Population  Total Population - Margin of Error  \
0             55200                                 NaN   
1            208107                                 NaN   
2             25782                                 NaN   
3             22527                                 NaN   
4             57645                                 NaN   

   Population Density (people per square kilometer)       created_user  \
0                                         35.853419  esri_demographics   
1                                         50.541504  esri_demographics   
2                                         11.247981  esri_demographics   
3                                         13.973114  esri_demographics   
4                                         34.515816  esri_demographics   

     created_date   last_edited_user last_edited_date Shape__Area  \
0  1/4/2020 20:53  esri_demographics   1/4/2020 20:53    2.066037   
1  1/4/2020 20:53  esri_demographics   1/4/2020 20:53    4.483746   
2  1/4/2020 20:53  esri_demographics   1/4/2020 20:53    2.695262   
3  1/4/2020 20:53  esri_demographics   1/4/2020 20:53    1.887514   
4  1/4/2020 20:53  esri_demographics   1/4/2020 20:53    2.423552   

  Shape__Length  Population Density - Margin of Error  
0      0.150256                                   NaN  
1      0.409904                                   NaN  
2      0.223270                                   NaN  
3      0.156473                                   NaN  
4      0.164405                                   NaN  

[5 rows x 22 columns]

In [170]:
housePop.describe()

COUNTYNS  Geographic Identifier - FIPS Code  \
count  3.220000e+03                        3220.000000   
mean   9.628897e+05                       31393.605280   
std    5.174236e+05                       16292.078954   
min    2.390100e+04                        1001.000000   
25%    4.849888e+05                       19032.500000   
50%    9.741585e+05                       30024.000000   
75%    1.419973e+06                       46105.500000   
max    2.516404e+06                       72153.000000   

       Area of Land (square meters)  Area of Water (square meters)  \
count                  3.220000e+03                   3.220000e+03   
mean                   2.844093e+09                   2.143251e+08   
std                    9.255389e+09                   1.220804e+09   
min                    5.300265e+06                   0.000000e+00   
25%                    1.085268e+09                   7.009854e+06   
50%                    1.565689e+09                   1.938601e+07   
75%                    2.368884e+09                   6.069241e+07   
max                    3.770000e+11                   2.599228e+10   

       Average Household Size  Average Household Size - Margin of Error  \
count             3220.000000                               3220.000000   
mean                 2.529093                                  0.081494   
std                  0.279374                                  0.067781   
min                  1.340000                                  0.010000   
25%                  2.350000                                  0.040000   
50%                  2.490000                                  0.070000   
75%                  2.650000                                  0.100000   
max                  4.970000                                  1.140000   

       Average Household Size of Owner-Occupied Unit  \
count                                    3219.000000   
mean                                        2.563088   
std                                         0.274885   
min                                         1.790000   
25%                                         2.400000   
50%                                         2.520000   
75%                                         2.690000   
max                                         4.880000   

       Average Household Size of Owner-Occupied Unit - Margin of Error  \
count                                        3219.000000                 
mean                                            0.102336                 
std                                             0.079810                 
min                                             0.010000                 
25%                                             0.050000                 
50%                                             0.090000                 
75%                                             0.130000                 
max                                             1.460000                 

       Average Household Size of Renter-Occupied Unit  \
count                                     3220.000000   
mean                                         2.444950   
std                                          0.393263   
min                                          1.360000   
25%                                          2.200000   
50%                                          2.400000   
75%                                          2.640000   
max                                          6.810000   

       Average Household Size of Renter-Occupied Unit - Margin of Error  \
count                                        3220.000000                  
mean                                            0.202314                  
std                                             0.182878                  
min                                             0.010000                  
25%                                             0.100000                  
50%                                          

In [171]:
healthInc = pd.read_csv('Health_Insurance_Coverage_-_Counties_2015-2019.csv')
healthInc.head()

STATEFP  COUNTYFP  COUNTYNS        AFFGEOID  GEOID County_Name  LSAD  \
0        1         1    161526  0500000US01001   1001     Autauga     6   
1        1         3    161527  0500000US01003   1003     Baldwin     6   
2        1         5    161528  0500000US01005   1005     Barbour     6   
3        1         7    161529  0500000US01007   1007        Bibb     6   
4        1         9    161530  0500000US01009   1009      Blount     6   

          ALAND      AWATER GEO_PARENT_NAME  ...  \
0  1.539602e+09    25706961         Alabama  ...   
1  4.117622e+09  1132980868         Alabama  ...   
2  2.292160e+09    50523213         Alabama  ...   
3  1.612167e+09     9602089         Alabama  ...   
4  1.670104e+09    15015467         Alabama  ...   

   Civilian noninstitutionalized population under 19 years  \
0                                              13915         
1                                              48427         
2                                               5594         
3                                               4759         
4                                              13974         

   Civilian noninstitutionalized population under 19 years *margin of error  \
0                                                166                          
1                                                382                          
2                                                 76                          
3                                                 77                          
4                                                151                          

   Total Civilian Noninstitutionalized Population under 19 yrs - No health insurance coverage  \
0                                                242                                            
1                                               1836                                            
2                                                184                                            
3                                                 94                                            
4                                                822                                            

   Total Civilian Noninstitutionalized Population under 19 yrs - No health insurance coverage *margin of error  \
0                                                131                                                             
1                                                537                                                             
2                                                110                                                             
3                                                 82                                                             
4                                                347                                                             

   Percent of Population with No Health Insurance Coverage  \
0                                                7.1         
1                                                8.9         
2                                               11.3         
3                                               10.7         
4                                               10.8         

   Percent of Population with No Health Insurance Coverage *margin of error  \
0                                                1.0                          
1                                                0.7                          
2                                                1.5                          
3                                                2.2                          
4                                                1.4                          

   Total Civilian Noninstitutionalized Population - No health insurance coverage  \
0                                               3856                               
1                                              18679                               
2                      

In [172]:
healthInc.describe()

STATEFP     COUNTYFP      COUNTYNS         GEOID         LSAD  \
count  3220.000000  3220.000000  3.220000e+03   3220.000000  3220.000000   
mean     31.290683   102.922050  9.628897e+05  31393.605280     6.569565   
std      16.277207   106.693846  5.174236e+05  16292.078954     2.652226   
min       1.000000     1.000000  2.390100e+04   1001.000000     0.000000   
25%      19.000000    35.000000  4.849888e+05  19032.500000     6.000000   
50%      30.000000    79.000000  9.741585e+05  30024.000000     6.000000   
75%      46.000000   133.000000  1.419973e+06  46105.500000     6.000000   
max      72.000000   840.000000  2.516404e+06  72153.000000    25.000000   

              ALAND        AWATER   POP_DENSITY  \
count  3.220000e+03  3.220000e+03   3220.000000   
mean   2.844436e+09  2.139761e+08    112.750562   
std    9.256066e+09  1.220470e+09    692.745183   
min    5.300262e+06  0.000000e+00      0.014261   
25%    1.085267e+09  7.004659e+06      6.683385   
50%    1.565686e+09  1.930200e+07     17.895454   
75%    2.368903e+09  6.022325e+07     51.768885   
max    3.770000e+11  2.597330e+10  27806.766950   

       Employer Insurance only - Pop < 19  \
count                         3220.000000   
mean                         11564.554969   
std                          36062.877369   
min                              1.000000   
25%                            944.250000   
50%                           2415.000000   
75%                           7056.500000   
max                         937617.000000   

       Employer Insurance only - Pop < 19 *margin of error  ...  \
count                                        3220.000000    ...   
mean                                          524.306211    ...   
std                                           605.854098    ...   
min                                             4.000000    ...   
25%                                           192.000000    ...   
50%                                           340.500000    ...   
75%                                           614.000000    ...   
max                                          8901.000000    ...   

       Civilian noninstitutionalized population under 19 years  \
count                                       3.220000e+03         
mean                                        2.435040e+04         
std                                         7.911533e+04         
min                                         1.000000e+00         
25%                                         2.590500e+03         
50%                                         6.118500e+03         
75%                                         1.579175e+04         
max                                         2.346589e+06         

       Civilian noninstitutionalized population under 19 years *margin of error  \
count                                        3220.000000                          
mean                                          162.774845                          
std                                           146.676597                          
min                                             4.000000                          
25%                                            72.000000                          
50%                                           121.000000                          
75%                                           195.000000                          
max                                          1712.000000                          

       Total Civilian Noninstitutionalized Population under 19 yrs - No health insurance coverage  \
count                                        3220.000000                                            
mean                                         1232.237888                                            
std                                          5043.523352                                            
min                                             0.000000                                            
25%     

In [173]:
incBen = pd.read_csv('Income_and_Benefits_-_Counties_2015-2019.csv')
incBen.head()

STATEFP  COUNTYFP  COUNTYNS        AFFGEOID  GEOID     NAME  LSAD  \
0        1         1    161526  0500000US01001   1001  Autauga     6   
1        1         3    161527  0500000US01003   1003  Baldwin     6   
2        1         5    161528  0500000US01005   1005  Barbour     6   
3        1         7    161529  0500000US01007   1007     Bibb     6   
4        1         9    161530  0500000US01009   1009   Blount     6   

          ALAND      AWATER GEO_PARENT_NAME  ...  \
0  1.539602e+09    25706961         Alabama  ...   
1  4.117622e+09  1132980868         Alabama  ...   
2  2.292160e+09    50523213         Alabama  ...   
3  1.612167e+09     9602089         Alabama  ...   
4  1.670104e+09    15015467         Alabama  ...   

   Total Households - With Supplemental Security Income  \
0                                               1302      
1                                               3568      
2                                               1019      
3                                                625      
4                                               1721      

   Total Households - With Supplemental Security Income *margin of error  \
0                                                235                       
1                                                408                       
2                                                184                       
3                                                183                       
4                                                286                       

   Total Households - With cash public assistance income  \
0                                                253       
1                                                910       
2                                                244       
3                                                148       
4                                                361       

   Total Households - With cash public assistance income *margin of error  \
0                                                102                        
1                                                264                        
2                                                 80                        
3                                                 91                        
4                                                137                        

   Total Households - With Food Stamp/SNAP benefits in the past 12 months  \
0                                               2746                        
1                                               6269                        
2                                               2509                        
3                                               1113                        
4                                               1977                        

   Total Households - With Food Stamp/SNAP benefits in the past 12 months *margin of error  \
0                                                407                                         
1                                                678                                         
2                                                238                                         
3                                                253                                         
4                                                264                                         

   Percent of Households with income in the past 12 months that was less than $75,000  \
0                                               60.9                                    
1                                               61.1                                    
2                                               80.1                                    
3                                               69.0                                    
4                                               70.1                                    

   Percent of Households with income in the past 12 months

In [174]:
incBen.describe()

STATEFP     COUNTYFP      COUNTYNS         GEOID         LSAD  \
count  3220.000000  3220.000000  3.220000e+03   3220.000000  3220.000000   
mean     31.290683   102.922050  9.628897e+05  31393.605280     6.569565   
std      16.277207   106.693846  5.174236e+05  16292.078954     2.652226   
min       1.000000     1.000000  2.390100e+04   1001.000000     0.000000   
25%      19.000000    35.000000  4.849888e+05  19032.500000     6.000000   
50%      30.000000    79.000000  9.741585e+05  30024.000000     6.000000   
75%      46.000000   133.000000  1.419973e+06  46105.500000     6.000000   
max      72.000000   840.000000  2.516404e+06  72153.000000    25.000000   

              ALAND        AWATER   POP_DENSITY  \
count  3.220000e+03  3.220000e+03   3220.000000   
mean   2.844537e+09  2.139761e+08    112.750562   
std    9.257767e+09  1.220470e+09    692.745183   
min    5.300262e+06  0.000000e+00      0.014261   
25%    1.085267e+09  7.004659e+06      6.683385   
50%    1.565686e+09  1.930200e+07     17.895454   
75%    2.368903e+09  6.022325e+07     51.768885   
max    3.770390e+11  2.597330e+10  27806.766950   

       Households: Receiving Food Stamps/SNAP (%)  \
count                                 3220.000000   
mean                                    13.553571   
std                                      8.156595   
min                                      0.000000   
25%                                      8.400000   
50%                                     12.200000   
75%                                     16.500000   
max                                     62.800000   

       Households: Receiving Food Stamps/SNAP (%) *margin of error  ...  \
count                                        3220.000000            ...   
mean                                            2.143696            ...   
std                                             1.362611            ...   
min                                             0.100000            ...   
25%                                             1.200000            ...   
50%                                             1.900000            ...   
75%                                             2.800000            ...   
max                                            19.100000            ...   

       Total Households - With Supplemental Security Income  \
count                                        3220.000000      
mean                                         2002.686025      
std                                          6644.036851      
min                                             0.000000      
25%                                           225.750000      
50%                                           615.500000      
75%                                          1522.750000      
max                                        223761.000000      

       Total Households - With Supplemental Security Income *margin of error  \
count                                         3220.00000                       
mean                                           204.53354                       
std                                            217.62440                       
min                                              2.00000                       
25%                                             74.00000                       
50%                                            150.00000                       
75%                                            251.00000                       
max                                           2663.00000                       

       Total Households - With cash public assistance income  \
count                                        3220.000000       
mean                                          915.497516       
std                                          3351.434522       
min                                             0.000000       
25%                                            81.000000       
50%                                    

In [175]:
occupSex = pd.read_csv('Occupation_by_Sex_-_Counties_2015-2019.csv')
occupSex.head()

STATEFP  COUNTYFP  COUNTYNS        AFFGEOID  GEOID     NAME  LSAD  \
0        1         1    161526  0500000US01001   1001  Autauga     6   
1        1         3    161527  0500000US01003   1003  Baldwin     6   
2        1         5    161528  0500000US01005   1005  Barbour     6   
3        1         7    161529  0500000US01007   1007     Bibb     6   
4        1         9    161530  0500000US01009   1009   Blount     6   

          ALAND      AWATER GEO_PARENT_NAME  ...  \
0  1.539602e+09    25706961         Alabama  ...   
1  4.117622e+09  1132980868         Alabama  ...   
2  2.292160e+09    50523213         Alabama  ...   
3  1.612167e+09     9602089         Alabama  ...   
4  1.670104e+09    15015467         Alabama  ...   

   Percent of Civilian Employed Population 16 Years and Over in Food Preparation and Serving Related Occupations  \
0                                                6.0                                                               
1                                                6.3                                                               
2                                                4.2                                                               
3                                                3.6                                                               
4                                                3.2                                                               

   Percent of Civilian Employed Population 16 Years and Over in Food Preparation and Serving Related Occupations*margin of error  \
0                                                1.4                                                                               
1                                                0.7                                                                               
2                                                1.3                                                                               
3                                                1.5                                                                               
4                                                0.9                                                                               

   Percent of Civilian Employed Population 16 Years and Over in Sales and Related Occupations  \
0                                               10.2                                            
1                                               13.1                                            
2                                                7.4                                            
3                                               11.5                                            
4                                                8.1                                            

   Percent of Civilian Employed Population 16 Years and Over in Sales and Related Occupations*margin of error  \
0                                                1.6                                                            
1                                                1.1                                                            
2                                                1.7                                                            
3                                                3.1                                                            
4                                                1.3                                                            

   Male - Natural resources, construction, and maintenance occupations  \
0                                               1882                     
1                                               8375                     
2                                               1122                     
3                                               1356                     
4                                               3428                     

   Percent of Civilian Employed Population 16 Years and Over

In [176]:
occupSex.describe()

STATEFP     COUNTYFP      COUNTYNS         GEOID         LSAD  \
count  3220.000000  3220.000000  3.220000e+03   3220.000000  3220.000000   
mean     31.290683   102.922050  9.628897e+05  31393.605280     6.569565   
std      16.277207   106.693846  5.174236e+05  16292.078954     2.652226   
min       1.000000     1.000000  2.390100e+04   1001.000000     0.000000   
25%      19.000000    35.000000  4.849888e+05  19032.500000     6.000000   
50%      30.000000    79.000000  9.741585e+05  30024.000000     6.000000   
75%      46.000000   133.000000  1.419973e+06  46105.500000     6.000000   
max      72.000000   840.000000  2.516404e+06  72153.000000    25.000000   

              ALAND        AWATER  \
count  3.220000e+03  3.220000e+03   
mean   2.844537e+09  2.139761e+08   
std    9.257767e+09  1.220470e+09   
min    5.300262e+06  0.000000e+00   
25%    1.085267e+09  7.004659e+06   
50%    1.565686e+09  1.930200e+07   
75%    2.368903e+09  6.022325e+07   
max    3.770390e+11  2.597330e+10   

       Total - Civilian employed population 16 years and over  \
count                                       3.220000e+03        
mean                                        4.840705e+04        
std                                         1.607070e+05        
min                                         3.300000e+01        
25%                                         4.608000e+03        
50%                                         1.075000e+04        
75%                                         2.948300e+04        
max                                         4.929863e+06        

       Total - Civilian employed population 16 years and over *margin of error  \
count                                        3220.000000                         
mean                                          638.408075                         
std                                           677.716564                         
min                                            11.000000                         
25%                                           253.000000                         
50%                                           437.000000                         
75%                                           748.250000                         
max                                          9150.000000                         

       Male - Civilian employed population 16 years and over  ...  \
count                                       3.220000e+03      ...   
mean                                        2.541800e+04      ...   
std                                         8.558493e+04      ...   
min                                         2.500000e+01      ...   
25%                                         2.469000e+03      ...   
50%                                         5.711500e+03      ...   
75%                                         1.558225e+04      ...   
max                                         2.663007e+06      ...   

       Percent of Civilian Employed Population 16 Years and Over in Food Preparation and Serving Related Occupations  \
count                                        3220.000000                                                               
mean                                            5.457453                                                               
std                                             1.830762                                                               
min                                             0.000000                                                               
25%                                             4.400000                                                               
50%                                             5.300000                                                               
75%                                             6.300000                                                               
max                                            20.100000                             

In [177]:
popPov = pd.read_csv('Population_and_Poverty_Status_-_Counties_2015-2019.csv')
popPov.head()

STATEFP  COUNTYFP  COUNTYNS        AFFGEOID  GEOID     NAME  LSAD  \
0        1         1    161526  0500000US01001   1001  Autauga     6   
1        1         3    161527  0500000US01003   1003  Baldwin     6   
2        1         5    161528  0500000US01005   1005  Barbour     6   
3        1         7    161529  0500000US01007   1007     Bibb     6   
4        1         9    161530  0500000US01009   1009   Blount     6   

          ALAND      AWATER GEO_PARENT_NAME  ...  \
0  1.539602e+09    25706961         Alabama  ...   
1  4.117622e+09  1132980868         Alabama  ...   
2  2.292160e+09    50523213         Alabama  ...   
3  1.612167e+09     9602089         Alabama  ...   
4  1.670104e+09    15015467         Alabama  ...   

   Percentage of people whose income in the past 12 months is below the poverty level - Related children of the householder under 5 years  \
0                                               21.2                                                                                        
1                                               16.0                                                                                        
2                                               59.0                                                                                        
3                                               17.3                                                                                        
4                                               21.0                                                                                        

   Percentage of people whose income in the past 12 months is below the poverty level - Related children of the householder under 5 years *margin of error  \
0                                                6.2                                                                                                         
1                                                3.7                                                                                                         
2                                                7.1                                                                                                         
3                                               12.6                                                                                                         
4                                                5.4                                                                                                         

   Percentage of people whose income in the past 12 months is below the poverty level - 65 years and over  \
0                                                8.7                                                        
1                                                7.4                                                        
2                                               16.8                                                        
3                                                6.9                                                        
4                                               10.9                                                        

   Percentage of people whose income in the past 12 months is below the poverty level - 65 years and over *margin of error  \
0                                                2.1                                                                         
1                                                1.4                                                                         
2                                                3.1                                                                         
3                                                3.2                                                                         
4                                                2.8                                                                         

   Percentage of people whose income in the past 

In [178]:
popPov.describe()

STATEFP     COUNTYFP      COUNTYNS         GEOID         LSAD  \
count  3220.000000  3220.000000  3.220000e+03   3220.000000  3220.000000   
mean     31.290683   102.922050  9.628897e+05  31393.605280     6.569565   
std      16.277207   106.693846  5.174236e+05  16292.078954     2.652226   
min       1.000000     1.000000  2.390100e+04   1001.000000     0.000000   
25%      19.000000    35.000000  4.849888e+05  19032.500000     6.000000   
50%      30.000000    79.000000  9.741585e+05  30024.000000     6.000000   
75%      46.000000   133.000000  1.419973e+06  46105.500000     6.000000   
max      72.000000   840.000000  2.516404e+06  72153.000000    25.000000   

              ALAND        AWATER   POP_DENSITY  \
count  3.220000e+03  3.220000e+03   3220.000000   
mean   2.844436e+09  2.139761e+08    112.750562   
std    9.256066e+09  1.220470e+09    692.745183   
min    5.300262e+06  0.000000e+00      0.014261   
25%    1.085267e+09  7.004659e+06      6.683385   
50%    1.565686e+09  1.930200e+07     17.895454   
75%    2.368903e+09  6.022325e+07     51.768885   
max    3.770000e+11  2.597330e+10  27806.766950   

       Total Households Below the Poverty Level  \
count                               3220.000000   
mean                                5014.081988   
std                                15831.216075   
min                                    2.000000   
25%                                  661.000000   
50%                                 1557.000000   
75%                                 3812.000000   
max                               481979.000000   

       Total Households Below the Poverty Level *margin of error  ...  \
count                                        3220.000000          ...   
mean                                          331.460870          ...   
std                                           335.315926          ...   
min                                             2.000000          ...   
25%                                           135.000000          ...   
50%                                           240.000000          ...   
75%                                           409.250000          ...   
max                                          4539.000000          ...   

       Percentage of people whose income in the past 12 months is below the poverty level - Related children of the householder under 5 years  \
count                                        3219.000000                                                                                        
mean                                           24.442591                                                                                        
std                                            14.312147                                                                                        
min                                             0.000000                                                                                        
25%                                            14.750000                                                                                        
50%                                            22.000000                                                                                        
75%                                            31.100000                                                                                        
max                                           100.000000                                                                                        

       Percentage of people whose income in the past 12 months is below the poverty level - Related children of the householder under 5 years *margin of error  \
count                                        3219.000000                                                                                                         
mean                                            9.207766                                                              

In [179]:
usa_covid = pd.read_csv('usa_county_27july.csv')
usa_covid.head()

UID iso2 iso3  code3  Geographic Identifier - FIPS Code County_Name  \
0  84001001   US  USA    840                             1001.0     Autauga   
1  84001003   US  USA    840                             1003.0     Baldwin   
2  84001005   US  USA    840                             1005.0     Barbour   
3  84001007   US  USA    840                             1007.0        Bibb   
4  84001009   US  USA    840                             1009.0      Blount   

  Province_State Country_Region        Lat      Long_          Combined_Key  \
0        Alabama             US  32.539527 -86.644082  Autauga, Alabama, US   
1        Alabama             US  30.727750 -87.722071  Baldwin, Alabama, US   
2        Alabama             US  31.868263 -85.387129  Barbour, Alabama, US   
3        Alabama             US  32.996421 -87.125115     Bibb, Alabama, US   
4        Alabama             US  33.982109 -86.567906   Blount, Alabama, US   

      Date  Confirmed  Deaths  
0  7/27/20        932      20  
1  7/27/20       2727      17  
2  7/27/20        529       4  
3  7/27/20        327       2  
4  7/27/20        601       1

In [180]:
usa_covid.describe()

UID        code3  Geographic Identifier - FIPS Code  \
count  3.340000e+03  3340.000000                        3330.000000   
mean   8.342958e+07   834.491617                       33061.684685   
std    4.315345e+06    36.498055                       18638.940791   
min    1.600000e+01    16.000000                          60.000000   
25%    8.401811e+07   840.000000                       19079.500000   
50%    8.402921e+07   840.000000                       31014.000000   
75%    8.404612e+07   840.000000                       47130.500000   
max    8.410000e+07   850.000000                       99999.000000   

               Lat        Long_      Confirmed        Deaths  
count  3340.000000  3340.000000    3340.000000   3340.000000  
mean     36.707212   -88.601474    1284.508683     44.314671  
std       9.062922    21.718982    6844.463318    448.443711  
min     -14.271000  -174.159600       0.000000      0.000000  
25%      33.895587   -97.790204      39.000000      0.000000  
50%      38.002344   -89.486710     151.000000      2.000000  
75%      41.573069   -82.311265     569.000000     12.000000  
max      69.314792   145.673900  224051.000000  23500.000000

As we can see, most of the data sets have 3,220 data points, while the Covid dataset has 3,340 data points. Looking into file, we noted why: the Covid dataset also included categories for travellers. People who travelled out of state and caught Covid were still registered, but under a different category than their home state. As such, there are some additional county data points which are not reflected in the other data sets. To enable us to perform a coherent study, we decided to ignore these data points for now, and will do a separate investigation on them later.

Now, we will start merging the data sets, one at a time using the pandas merge command, using the GNIS County Code as the key. (FOLLOW UP)

In [181]:
county_covid = pd.merge(housePop, healthInc, on="COUNTYNS")

In [182]:
county_covid = pd.merge(county_covid, incBen, on="COUNTYNS")

In [183]:
county_covid = pd.merge(county_covid, occupSex, on="COUNTYNS")

In [184]:
county_covid = pd.merge(county_covid, popPov, on="COUNTYNS")

In [185]:
#county_covid.to_csv('county_megaset.csv', index=False)

In [186]:
county_covid['Geographic Identifier - FIPS Code'].describe()

count     3220.000000
mean     31393.605280
std      16292.078954
min       1001.000000
25%      19032.500000
50%      30024.000000
75%      46105.500000
max      72153.000000
Name: Geographic Identifier - FIPS Code, dtype: float64

Since we are merging similar datasets, some of the columns are identical, to avoid future complications, we will remove those duplicates first.

In [187]:
county_covid = county_covid.loc[:,~county_covid.columns.duplicated()]

In [188]:
county_covid = pd.merge(county_covid, usa_covid, on="Geographic Identifier - FIPS Code")

In [189]:
county_covid.describe()

COUNTYNS  Geographic Identifier - FIPS Code  \
count  3.220000e+03                        3220.000000   
mean   9.628897e+05                       31393.605280   
std    5.174236e+05                       16292.078954   
min    2.390100e+04                        1001.000000   
25%    4.849888e+05                       19032.500000   
50%    9.741585e+05                       30024.000000   
75%    1.419973e+06                       46105.500000   
max    2.516404e+06                       72153.000000   

       Area of Land (square meters)  Area of Water (square meters)  \
count                  3.220000e+03                   3.220000e+03   
mean                   2.844093e+09                   2.143251e+08   
std                    9.255389e+09                   1.220804e+09   
min                    5.300265e+06                   0.000000e+00   
25%                    1.085268e+09                   7.009854e+06   
50%                    1.565689e+09                   1.938601e+07   
75%                    2.368884e+09                   6.069241e+07   
max                    3.770000e+11                   2.599228e+10   

       Average Household Size  Average Household Size - Margin of Error  \
count             3220.000000                               3220.000000   
mean                 2.529093                                  0.081494   
std                  0.279374                                  0.067781   
min                  1.340000                                  0.010000   
25%                  2.350000                                  0.040000   
50%                  2.490000                                  0.070000   
75%                  2.650000                                  0.100000   
max                  4.970000                                  1.140000   

       Average Household Size of Owner-Occupied Unit  \
count                                    3219.000000   
mean                                        2.563088   
std                                         0.274885   
min                                         1.790000   
25%                                         2.400000   
50%                                         2.520000   
75%                                         2.690000   
max                                         4.880000   

       Average Household Size of Owner-Occupied Unit - Margin of Error  \
count                                        3219.000000                 
mean                                            0.102336                 
std                                             0.079810                 
min                                             0.010000                 
25%                                             0.050000                 
50%                                             0.090000                 
75%                                             0.130000                 
max                                             1.460000                 

       Average Household Size of Renter-Occupied Unit  \
count                                     3220.000000   
mean                                         2.444950   
std                                          0.393263   
min                                          1.360000   
25%                                          2.200000   
50%                                          2.400000   
75%                                          2.640000   
max                                          6.810000   

       Average Household Size of Renter-Occupied Unit - Margin of Error  ...  \
count                                        3220.000000                 ...   
mean                                            0.202314                 ...   
std                                             0.182878                 ...   
min                                             0.010000                 ...   
25%                                             0.100000                 ...   
50%            

In [190]:
county_covid.to_csv('county_megaset.csv', index=False)

Now, we have the data set that we want to work with, which we will then conduct our Exploratory Data Analysis.